test/cpp_extensions/cpp_c10d_extension.cpp

```diff
@@ -0,0 +1,113 @@
+  return std::make_shared<ProcessGroupC10dTest::WorkC10dTest>();
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::allreduce_coalesced(
+      std::vector<at::Tensor>& tensors,
+      const AllreduceCoalescedOptions& opts) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support allreduce_coalesced");
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::reduce(
+    std::vector<at::Tensor>& tensors,
+    const ReduceOptions& opts) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support reduce");
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::allgather(
+    std::vector<std::vector<at::Tensor>>& outputTensors,
+    std::vector<at::Tensor>& inputTensors,
+    const AllgatherOptions& opts) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support allgather");
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::barrier(
+    const BarrierOptions& opts) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support barrier");
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::gather(
+    std::vector<std::vector<at::Tensor>>& outputTensors,
+    std::vector<at::Tensor>& inputTensors,
+    const GatherOptions& opts) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support gather");
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::scatter(
+    std::vector<at::Tensor>& outputTensors,
+    std::vector<std::vector<at::Tensor>>& inputTensors,
+    const ScatterOptions& opts) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support scatter");
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::reduce_scatter(
+    std::vector<at::Tensor>& outputTensors,
+    std::vector<std::vector<at::Tensor>>& inputTensors,
+    const ReduceScatterOptions& opts) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support reduce_scatter");
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::send(
+    std::vector<at::Tensor>& tensors,
+    int dstRank,
+    int tag) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support send");
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::recv(
+    std::vector<at::Tensor>& tensors,
+    int srcRank,
+    int tag) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support recv");
+}
+
+std::shared_ptr<ProcessGroup::Work> ProcessGroupC10dTest::recvAnysource(
+    std::vector<at::Tensor>& tensor,
+    int tag) {
+  throw std::runtime_error("ProcessGroupC10dTest does not support recvAnysource");
+}
+
+std::shared_ptr<ProcessGroup> ProcessGroupC10dTest::createProcessGroupC10dTest(
+    const std::shared_ptr<::c10d::Store>& store,
+    int rank,
+    int size,
+    const std::string& groupName = "") {
+  return std::make_shared<ProcessGroupC10dTest>(rank, size);
+}
+
+PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
+  m.def("ProcessGroupTEST", &ProcessGroupC10dTest::createProcessGroupC10dTest);
```

Comment:
Why all caps? Let's define this as `ProcessGroupTest`, both in pybind, as well as the class itself.